### DISCUSSION TITLE
Scroll synchronizing is broken (hacky)

### DISCUSSION THREAD

**Discussion by Chris2011** [moacirrf/netbeans-markdown#38] [Ideas]
With latest pre-release version 0.0.5-SNAPSHOT, the scrolling is broken. When I scroll in one pane, it is not synchronized anymore, but when I move my mouse to the other pane, the scrolled pane will be reset to that where the other pane is.

![scrolling-broken](https://user-images.githubusercontent.com/795658/205465140-59441e6d-8ea9-43ec-a0ec-32d2023cc428.gif)

Unfortunately it is very hacky and the version before was better. I tested it and I can understand your intention, but just test a real markdown file for example here and you will see what I mean: https://github.com/Chris2011/netbeans-textmate-files/blob/master/README.md. Fluid scrollen is better than snapping what I think was your intention.

**Comment by Chris2011** [comment]
Let us switch to a discussion to maybe find a better solution for that :)

**Reply by moacirrf** [reply]
Let's go, i converted to a discussion.

**Comment by moacirrf** [comment]
Can you explain this "fluid scrollen" idea a little more?
thanks

**Reply by Chris2011** [reply]
I just mean that "fluid scrolling" is just nornal scrolling it should not jump it should just scroll in a normal way. Maybe you can check my markdown file in VS Code and how it looks there. Maybe a look inside of the markdown viewer of VS Code can have a solution for this? Doesn't matter it was written in TS or not, just for the idea how it is working there.

**Comment by moacirrf** [comment]
This is the generated HTML of Flexmark, maybe can help.
[teste.txt](https://github.com/moacirrf/netbeans-markdown/files/10147593/teste.txt)

**Comment by moacirrf** [comment]
> With latest pre-release version 0.0.5-SNAPSHOT, the scrolling is broken. When I scroll in one pane, it is not synchronized anymore, but when I move my mouse to the other pane, the scrolled pane will be reset to that where the other pane is.
> 
> ![scrolling-broken](https://user-images.githubusercontent.com/795658/205465140-59441e6d-8ea9-43ec-a0ec-32d2023cc428.gif) [ ![scrolling-broken](https://user-images.githubusercontent.com/795658/205465140-59441e6d-8ea9-43ec-a0ec-32d2023cc428.gif) ](https://user-images.githubusercontent.com/795658/205465140-59441e6d-8ea9-43ec-a0ec-32d2023cc428.gif) [ ](https://user-images.githubusercontent.com/795658/205465140-59441e6d-8ea9-43ec-a0ec-32d2023cc428.gif)
> 
> Unfortunately it is very hacky and the version before was better. I tested it and I can understand your intention, but just test a real markdown file for example here and you will see what I mean: https://github.com/Chris2011/netbeans-textmate-files/blob/master/README.md. Fluid scrollen is better than snapping what I think was your intention.

I found the reset problem, its because  that i include a delay before start the scroll, when you scroll and change the focus to another panel, this reset occurs. Crazy!

**Comment by moacirrf** [comment]
I made some more fixes based on your issues, hope it help.
https://github.com/moacirrf/netbeans-markdown/releases/tag/latest

**Reply by Chris2011** [reply]
I missed it somehow and I tried the latest one from last month and it is still the same, it is scrolling to some ankers I think and this is not really intuitiv. Often it jumps over the point, where I scrolled to.

![anker-based-jumping](https://user-images.githubusercontent.com/795658/224178599-914a8334-c0cb-491f-9ef8-4167600aa0b2.gif)

**Reply by moacirrf** [reply]
I don't have a solution for that until the moment.

**Reply by Chris2011** [reply]
Ok no problem, amybe it would help to check how VS Code handles it. We can debug it via the dev tools inside VS Code mayb that helps a bit. Reverse engineering :D

**Reply by moacirrf** [reply]
> Ok no problem, amybe it would help to check how VS Code handles it. We can debug it via the dev tools inside VS Code mayb that helps a bit. Reverse engineering :D
1-VS Code
2- Eclipse
3- Intelij
We have many guys to "Reverse engineering", when i have a little more time i will try that. LOL
thanks
